fontforge/svg.c

```diff
@@ -2777,7 +2777,36 @@ return( NULL );
 return( eret );
 }
 
-static Entity *SVGParseSVG(xmlNodePtr svg,int em_size,int ascent,bool scale) {
+void SCDimensionFromSVG(xmlNodePtr svg, SplineChar *sc, bool vert) {
```

Comment:
No viewBox handling needed in this change, Frank?

Reply:
I don't know how we would map the viewBox values to glyph state.

Reply:
I think @ctrlcctrlv was thinking that given that the viewbox is x, y, width, height, one could consider the third and fourth values as alternate sources of width and height parameters.

Reply:
Thank you, that's right.

Reply:
Is there any case in which the width/height attribute would be missing or inapplicable (due to non-absolute units) and the viewBox would be useful? For what it is worth, the specification seems to suggest [here](https://www.w3.org/TR/SVG2/geometry.html#Sizing) that width and height are at least always present.

Reply:
I think values of "100%" are common, and I think I've seen elsewhere that "100%" is the implicit default for the svg element, but that could be wrong. 

But yes, if the width element is a percentage or a length measure and the viewBox is present then the latter is a good source of a width dimension. That's basically what another client is going to do: treat the boundaries specified by the viewBox as the abstract dimensions of the SVG and then scale in one or both dimensions according to the length and width percentages or lengths.

Reply:
I looked at an Inkscape file, since those are our most likely import candidates. Inkscape uses the user-specified page unit in the width and height fields ("8.5in" for example), with the viewBox in the user-specified drawing unit without a suffix. The patch as it is thus only works for a narrow range of inputs. But doing this right is not trivial, and I want to get what we have merged for the benefit of the weather symbols project.

Using the viewBox seems like a good fallback, but I do wonder whether we need to devote any attention to the case of one of the first two values being non-zero.

Reply:
I believe the viewbox is x, y (starting point), width, height, so the first two values should be non-relevant for these purposes.